src/WebJobs.Extensions.DurableTask/DurableOrchestrationClient.cs

```diff
@@ -155,6 +156,22 @@ public override HttpManagementPayload CreateHttpManagementPayload(string instanc
             return await this.GetDurableOrchestrationStatusAsync(state, showHistory, showHistoryOutput);
         }
 
+        /// <inheritdoc />
+        public override async Task<IList<DurableOrchestrationStatus>> GetStatusAsync(bool showHistory = false, bool showHistoryOutput = false, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            // TODO this cast is to avoid to change DurableTask.Core. Change it to use TaskHubClient.
+            var serviceClient = (DurableTask.AzureStorage.AzureStorageOrchestrationService)this.client.serviceClient;
+            var states = await serviceClient.GetOrchestrationStateAsync(cancellationToken);
+            var tasks = new List<Task<DurableOrchestrationStatus>>();
+            foreach (var state in states)
+            {
+                tasks.Add(this.GetDurableOrchestrationStatusAsync(state, showHistory, showHistoryOutput));
```

Comment:
I worry that this could cause thread-pool starvation because we're potentially starting an unlimited number of tasks in parallel. I looked at the implementation of `GetDurableOrchestrationStatusAsync`, and it appears that it only does async work if `showHistory` is true. Otherwise, it completes quickly and synchronously. To avoid risk, can we remove `showHistory` and `showHistoryOutput` from this new API and set it always to `false` when calling `GetDurableOrchestrationStatusAsync`? I think getting the history of every instance at the same time will not be necessary for the DevOps scenarios we are interested in, and this will make sure we do not explode our thread usage.

If we make this suggested change, then I think you can get rid of `Task.WhenAll` and just add items to the list one at a time.